docs/guide/topics/webhooks.rst

```diff
@@ -0,0 +1,224 @@
+
+It is worth noting that for asynchronous :class:`Webhook`, The ``session`` parameter
+takes in a :class:`aiohttp.ClientSession` instance. However for synchronous variant, The
+``session`` parameter takes the :class:`requests.Session` instance.
+
+:meth:`~Webhook.from_url` is a helper that automatically extracts webhook ID and token
+from the given URL. We also have :meth:`Webhook.partial` method to initialize partial
+webhooks directly using the ID and token.
+
+.. code-block:: py
+
+    async with aiohttp.ClientSession() as session:
+        webhook = discord.Webhook.partial(webhook_id, webhook_token, session=session)
+
+Fetching webhooks
+~~~~~~~~~~~~~~~~~
+
+Partial webhooks only contain the webhook ID and token as these are the only required
+components to perform basic requests. However if you want to retrieve complete webhook
+information, you can fetch them. Use the :meth:`Webhook.fetch` method to do so.
+
+.. code-block:: py
+
+    partial_webhook = discord.Webhook.from_url("url-here", session=session)
+    print(partial_webhook.is_partial()) # True
+    webhook = await partial_webhook.fetch()
+    print(webhook.is_partial()) # False
+
+When you have a :meth:`Client` instance available, you can use it's :meth:`~Client.fetch_webhook` method to
+fetch the webhook using the ID. It is worth noting that this method does not require you to create a separate 
+HTTP session. The library's internal session is automatically bound to the webhook and is managed by the library.
+
+Fetched webhook will include the complete webhook information like username, avatar etc. Fetching the webhook isn't
+necessary when you just want to perform simple HTTP operations such as sending messages, editing or deleting them.
+Only fetch the webhook when you need this information, avoid making unnecessary API calls.
+
+Sending messages
+~~~~~~~~~~~~~~~~
+
+This is where the fun begins, Let's start posting messages through our webhook.
+We use the :meth:`Webhook.send` method to send the messages.
+
+.. code-block:: py
+
+    async with aiohttp.ClientSession() as session:
+        webhook = discord.Webhook.from_url("webhook-url", session=session)
+        await webhook.send("Hello World")
+
+This will produce the following output in Discord:
+
+.. image:: /images/guide/webhooks/webhook_message.png
+    :alt: Webhook message
+
+Like normal messages, Webhooks support embeds, file attachments, allowed mentions and
+other message features. An example of sending embeds is given below.
+
+.. code-block:: py
+
+    async with aiohttp.ClientSession() as session:
+        webhook = discord.Webhook.from_url("webhook-url", session=session)
+        embed = discord.Embed(title="Hello World", description="I'm a nice embed!")
+        await webhook.send(embed=embed)
+
+.. image:: /images/guide/webhooks/webhook_message_with_embed.png
+    :alt: Webhook message with embed.
+
+There are some features that are exclusive to webhook messages like hyperlinks in
+raw message's content.
```

Comment:
This has changed relatively recently and is now possible within normal messages.